converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/ChangeFeedProcessor.cs

```diff
@@ -64,72 +58,77 @@ public async Task ProcessAsync(TimeSpan pollIntervalDuringCatchup, CancellationT
+                if (changeFeedEntries.Count == 0 && latest == state.SyncedSequence)
                 {
                     _logger.LogInformation("No new DICOM events to process.");
-
                     return;
                 }
 
-                long maxSequence = changeFeedEntries[^1].Sequence;
-
-                // Process each change feed as a FHIR transaction.
-                foreach (ChangeFeedEntry changeFeedEntry in changeFeedEntries)
-                {
-                    try
-                    {
-                        if (!(changeFeedEntry.Action == ChangeFeedAction.Create && changeFeedEntry.State == ChangeFeedState.Deleted))
-                        {
-                            await _fhirTransactionPipeline.ProcessAsync(changeFeedEntry, cancellationToken);
-                            _logger.LogInformation("Successfully processed DICOM event with SequenceID: {SequenceId}", changeFeedEntry.Sequence);
-                        }
-                        else
-                        {
-                            _logger.LogInformation("Skip DICOM event with SequenceId {SequenceId} due to deletion before processing creation.", changeFeedEntry.Sequence);
-                        }
-                    }
-                    catch (Exception ex) when (ex is FhirNonRetryableException or DicomTagException or TimeoutRejectedException)
-                    {
-                        string studyInstanceUid = changeFeedEntry.StudyInstanceUid;
-                        string seriesInstanceUid = changeFeedEntry.SeriesInstanceUid;
-                        string sopInstanceUid = changeFeedEntry.SopInstanceUid;
-                        long changeFeedSequence = changeFeedEntry.Sequence;
-
-                        ErrorType errorType = ErrorType.FhirError;
-
-                        if (ex is DicomTagException)
-                        {
-                            errorType = ErrorType.DicomError;
-                        }
-                        else if (ex is TimeoutRejectedException)
-                        {
-                            errorType = ErrorType.TransientFailure;
-                        }
-
-                        await _exceptionStore.WriteExceptionAsync(
-                            changeFeedEntry,
-                            ex,
-                            errorType,
-                            cancellationToken);
-
-                        _logger.LogError("Failed to process DICOM event with SequenceID: {SequenceId}, StudyUid: {StudyInstanceUid}, SeriesUid: {SeriesInstanceUid}, instanceUid: {SopInstanceUid}  and will not be retried further. Continuing to next event.", changeFeedEntry.Sequence, studyInstanceUid, seriesInstanceUid, sopInstanceUid);
-                    }
-                }
+                // Otherwise, process any new entries and increment the sequence
+                long maxSequence = changeFeedEntries.Count > 0 ? changeFeedEntries[^1].Sequence : state.SyncedSequence + DefaultLimit;
```

Comment:
`LastOrDefault` would actually be slower and costlier, as it would require allocating an `IEnumerator` and enumerating the results until the end of the sequence. Although practically, it probably isn't too much worse as:
1. The source code is probably optimized for indexed collections such that it _probably_ casts the `IEnumerable<T>` to like `IList<T>`
2. The collection will only have 10 elements at most